sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -336,22 +347,38 @@
      * of how the partitions are distributed among the splits.
      */
     public Read<K, V> withTopic(String topic) {
-      return withTopics(ImmutableList.of(topic));
+      return withTopics(commaSeparatedStrings(ValueProvider.StaticValueProvider.of(topic)));
+    }
+
+    /**
+     * Like above but with a {@link ValueProvider ValueProvider&lt;String&gt;}.
+     */
+    public Read<K, V> withTopic(ValueProvider<String> topic) {
+      return withTopics(commaSeparatedStrings(topic));
     }
 
     /**
-     * Sets a list of topics to read from. All the partitions from each
-     * of the topics are read.
+     * Sets a list of topics to read from with a {@link List List&lt;String&gt;}.
+     * All the partitions from each of the topics are read.
      *
      * <p>See {@link KafkaUnboundedSource#split(int, PipelineOptions)} for description
      * of how the partitions are distributed among the splits.
      */
     public Read<K, V> withTopics(List<String> topics) {
-      checkState(
-          getTopicPartitions().isEmpty(), "Only topics or topicPartitions can be set, not both");
-      return toBuilder().setTopics(ImmutableList.copyOf(topics)).build();
+      return withTopics(ValueProvider.StaticValueProvider.of(topics));
     }
 
+    /**
+     * Like above but with a {@link ValueProvider ValueProvider&lt;List&lt;String&gt;&gt;}.
+     *
+     */
+    public Read<K, V> withTopics(ValueProvider<List<String>> topics) {
+      checkArgument(
+              getTopicPartitions().isAccessible(),
```

Comment:
I am new to Guava, I wanted to check if getTopicPartitions() is not provided because Topics and TopicPartitions are mutually exclusive options. I need to do this validation at graph time if possible. Can you please guide me?